odbc/src/api/bind_col_tests.rs

```diff
@@ -0,0 +1,319 @@
+            let _ = Box::from_raw(buffer as *mut WChar);
+        }
+    }
+
+    #[test]
+    fn test_unbinding_column() {
+        // Set up MongoHandle
+        let env = &mut MongoHandle::Env(Env::with_state(EnvState::Allocated));
+        let conn =
+            &mut MongoHandle::Connection(Connection::with_state(env, ConnectionState::Allocated));
+        let stmt: *mut _ =
+            &mut MongoHandle::Statement(Statement::with_state(conn, StatementState::Allocated));
+
+        unsafe {
+            // Get Statement
+            let s = (*stmt).as_statement().unwrap();
+
+            // Set the bound_cols to non-None initially.
+            *s.bound_cols.write().unwrap() = Some(map! {
+                1 => BoundColInfo {
+                    target_type: CDataType::SQL_C_SLONG as SmallInt,
+                    target_buffer: null_mut(),
+                    buffer_length: 1,
+                    length_or_indicator: null_mut(),
+                }
+            });
+
+            // set all statement attributes to the correct values.
+            s.attributes.write().unwrap().row_bind_offset_ptr = null_mut();
+            s.attributes.write().unwrap().row_array_size = 1;
+            s.attributes.write().unwrap().row_bind_type = BindType::SQL_BIND_BY_COLUMN as usize;
+
+            // Set the mongo_statement to have non-empty cursor initially.
+            // Here, we create a MockQuery with nonsense dummy data since the
+            // values themselves do not matter.
+            let mock_query = &mut MongoQuery::new(
+                vec![doc! {"x": "y"}, doc! {"x": "z"}],
+                vec![
+                    MongoColMetadata::new(
+                        "test_db",
+                        "dn".to_string(),
+                        "fn".to_string(),
+                        Schema::Atomic(Atomic::Scalar(BsonTypeName::Int)),
+                        Nullability::SQL_NO_NULLS,
+                        TypeMode::Simple,
+                    ),
+                    MongoColMetadata::new(
+                        "test_db",
+                        "dn".to_string(),
+                        "fn".to_string(),
+                        Schema::Atomic(Atomic::Scalar(BsonTypeName::Int)),
+                        Nullability::SQL_NO_NULLS,
+                        TypeMode::Simple,
+                    ),
+                ],
+            );
+
+            // Must call next to set the `current` field.
+            let _ = mock_query.next(None);
+
+            // Set the mongo_statement
+            *s.mongo_statement.write().unwrap() = Some(Box::new(mock_query.clone()));
+
+            let indicator: *mut Len = null_mut();
+
+            // Assert that SQLBindCol is successful
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLBindCol(
+                    stmt as *mut _,
+                    1,
+                    CDataType::SQL_C_SLONG as SmallInt,
+                    null_mut(),
```

Comment:
```suggestion
                    null_mut(), // Note that when TargetValuePtr is null, the driver unbinds the data buffer for the column
```
Let's include a comment explaining that _this_ is the most consequential part of the test.